packages/gatsby-theme-docs/gatsby-node.js

```diff
@@ -62,6 +63,11 @@ const resolverPassthrough = ({
   const result = await field.resolve(resolveNode(mdxNode), args, context, {
     fieldName,
   });
+
+  if (fieldName === 'tableOfContents') {
+    return processTableOfContentFields(result);
+  }
```

Comment:
Nitpick: the `resolverPassthrough` is a generic function, we shouldn't put specific logic into it.

Instead we can add a new option `processResult` that can be passed on demand.

```js
const resolverPassthrough = ({
  typeName = 'Mdx',
  fieldName,
  resolveNode = identity,
  processResult = identity,
}) => async (source, args, context, info) => {
  // ...
  return processResult(result);
}
```

Then in the `tableOfContents` schema resolver, you use it like

```js
resolve: resolverPassthrough({
  fieldName: 'tableOfContents',
  processResult: processTableOfContentFields,
})
```